# Load and Inspect Tg Dataset

This notebook loads the raw polymer Tg dataset and inspects its basic structure,
including dataset size and column names.

In [4]:
import pandas as pd

In [6]:
df = pd.read_csv("../data/raw/tg_dataset.csv")

In [7]:
df.shape

(7284, 4)

In [8]:
df.columns

Index(['SMILES', 'Tg', 'PID', 'Polymer Class'], dtype='object')

In [ ]:
print(df.shape)
print(df.columns)

(7284, 4)
Index(['SMILES', 'Tg', 'PID', 'Polymer Class'], dtype='object')


In [15]:
missing_summary = df.isnull().sum()
print(missing_summary)

SMILES           0
Tg               0
PID              0
Polymer Class    0
dtype: int64


In [16]:
print(df["Tg"].describe())

count    7284.000000
mean      140.190628
std       112.961182
min      -139.000000
25%        53.112500
50%       132.000000
75%       230.000000
max       495.000000
Name: Tg, dtype: float64


In [18]:
duplicate_count = df["SMILES"].duplicated().sum()
print(f"Duplicate SMILES: {duplicate_count}")

Duplicate SMILES: 110


In [20]:
duplicate_rows = df[df["SMILES"].duplicated(keep=False)]
duplicate_rows

SMILES          Tg      PID   Polymer Class
0                              *C*  -54.000000  P010001     Polyolefins
30                      */C=C/CCC*  -99.600000  P010068      Polydienes
31                      */C=C/CCC*  -99.600000  P010082      Polydienes
32                      */C=C/CCC*  -99.600000  P010098      Polydienes
33                       */C=C/CC*  -69.000000  P010070      Polydienes
...                            ...         ...      ...             ...
7231  */C=C/c1cc(OCCCCCC)c(*)cc1OC   60.000000  P512152  Polyphenylenes
7251  */C=C/c1cc(OCCCCCC)c(*)cc1OC   60.000000  P512089  Polyphenylenes
7252  */C=C/c1cc(OCCCCCC)c(*)cc1OC   60.000000  P512152  Polyphenylenes
7282                       */C=C/*  332.708838  P014362      Polydienes
7283                           *C*  270.623317  P010001     Polyolefins

[143 rows x 4 columns]

### Observations
- 7284 data entry, 4 columns
- No missing values.
- 110 duplicated SMILES entries were identified.